client/src/app/queries/questionnaires.ts

```diff
@@ -0,0 +1,108 @@
+import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
+import { AxiosError } from "axios";
+
+// TODO Uncomment when Hub API is ready
+import {
+  deleteQuestionnaire,
+  getQuestionnaires,
+  updateQuestionnaire,
+} from "@app/api/rest";
+import { Questionnaire } from "@app/api/models";
+
+export const QuestionnairesTasksQueryKey = "questionnaires";
+
+// TODO Replace when Hub API is ready
+export const useFetchQuestionnaires = (mockQuestionnaires: Questionnaire[]) => {
+  return {
+    questionnaires: mockQuestionnaires,
+    isFetching: false,
+    fetchError: null,
+  };
+};
+
+// export const useFetchQuestionnaires = () => {
+//   const { isLoading, data, error } = useQuery({
+//     queryKey: [QuestionnairesTasksQueryKey],
+//     queryFn: getQuestionnaires,
+//     onError: (error) => console.log("error, ", error),
+//   });
+//   return {
+//     questionnaires: data || [],
+//     isFetching: isLoading,
+//     fetchError: error as AxiosErr_event
+// TODO Replace when Hub API is ready
+const mockUpdateQuestionnaire = (
+  obj: Questionnaire,
+  mockQuestionnaires: Questionnaire[],
+  setMockQuestionnaires: (questionnaires: Questionnaire[]) => void
+) => {
+  const newMockQuestionnaires = mockQuestionnaires.map((questionnaire) =>
+    questionnaire.id === obj.id ? obj : questionnaire
+  );
+  setMockQuestionnaires(newMockQuestionnaires);
+};
+
+export const useUpdateQuestionnaireMutation = (
+  _onSuccess: () => void,
+  _onError: (err: AxiosError) => void
+) => {
+  return {
+    mutationFn: mockUpdateQuestionnaire,
+  };
+};
+
+// export const useUpdateQuestionnaireMutation = (
+//   onSuccess: () => void,
+//   onError: (err: AxiosError) => void
+// ) => {
+//   const queryClient = useQueryClient();
+//   return useMutation({
+//     mutationFn: updateQuestionnaire,
+//     onSuccess: () => {
+//       onSuccess();
+//       queryClient.invalidateQueries([QuestionnairesTasksQueryKey]);
+//     },
+//     onError: onError,
+//   });
+// };
+
+// TODO Replace when Hub API is ready
+const mockDeleteQuestionnaire = (
+  id: number,
+  mockQuestionnaires: Questionnaire[],
+  setMockQuestionnaires: (questionnaires: Questionnaire[]) => void
+) => {
+  const newMockQuestionnaires = mockQuestionnaires.filter(
+    (questionnaire) => questionnaire.id !== id
+  );
+  setMockQuestionnaires(newMockQuestionnaires);
+};
+
+export const useDeleteQuestionnaireMutation = (
+  _onSuccess: (name: string) => void,
```

Comment:
Why add the underscores here?

Reply:
This is mock stuff

Reply:
Should probably name is accordingly, then. Looks like a real mutation function we would expect to use. 

Are underscores standard for mock function param names?

Reply:
+1 if the underscores are there just to remind us this is mock stuff, we may want to instead use the real names and add a TODO comment instead.